File: go/src/playground/testdata/src/ping/ping.go
// Copyright 2015 The Vanadium Authors. All rights reserved.
// Use of this source code is governed by a BSD-style
// license that can be found in the LICENSE file.

// +build ignore

package main

import (
	"fmt"

	"v.io/v23"
	"v.io/x/lib/vlog"
	_ "v.io/x/ref/profiles"

	"pingpong"
)

func main() {
	ctx, shutdown := v23.Init()
	defer shutdown()

	s := pingpong.PingPongClient("pingpong")

	fmt.Printf("Pinging\n")
	pong, err := s.Ping(ctx, "PING")
	if err != nil {
		vlog.Fatal("error pinging: ", err)
	}
	fmt.Println(pong)
}

File: go/src/playground/testdata/src/ping/ping.js
var vanadium = require('vanadium');

vanadium.init(function(err, rt) {
  if (err) throw err;

  var ctx = rt.getContext();

  console.log('Binding to service');
  rt.newClient().bindTo(ctx, 'pingpong', function(err, s) {
    if (err) throw err;

    console.log('Pinging');
    s.ping(ctx, 'PING', function(err, pong) {
      if (err) throw err;

      console.log(pong);
      process.exit(0);
    });
  });
});
